src/Bicep.Core/Diagnostics/DiagnosticBuilder.cs

```diff
@@ -442,6 +442,11 @@ public ErrorDiagnostic ArgumentCountMismatch(int argumentCount, int mininumArgum
                 DiagnosticLevel.Warning,
                 "BCP081",
                 $"Resource type {resourceTypeReference.FormatName()} does not have types available");
+
+            public ErrorDiagnostic SymbolicNameCannotUseReservedNamespaceName(string name, IEnumerable<string> namespaces) => new ErrorDiagnostic(
+                TextSpan,
+                "BCP082",
+                $"The symbolic name '{name}' cannot reference a reserved namespace. Reserved namespaces are '{string.Join(", ", namespaces)}'.");
```

Comment:
Should the string.Join include single quotes? Otherwise this is presumably going to look like `'az, sys'` rather than `'az', 'sys'`. Might also be worth creating a static helper method for this as I think there are a few other places in this file where we're doing something similar.